/* Copyright (C) 2012 Mike Stitt
#
# Permission is hereby granted, free of charge, to any person obtaining a copy of
# this software and associated documentation files (the "Software"), to deal in
# the Software without restriction, including without limitation the rights to
# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
# of the Software, and to permit persons to whom the Software is furnished to do
# so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
#
*/

package testCapture;

message colorFilterSettings {
  required int32 minHue = 1;
  required int32 maxHue = 2;
  required int32 minSat = 3;
  required int32 maxSat = 4;
  required int32 minVal = 5;
  required int32 maxVal = 6;
  required int32 beatPeriod = 7;
}

message findPolySettings {
  required int32 cannyThreshold1 = 1;
  required int32 cannyThreshold2 = 2;
  required int32 cannyAperature = 3;
  required int32 minPerimeter = 4;
  required int32 minArea = 5;
  required int32 deviationRatio = 7;
}

message captureSettings {
  required int32               magicTest = 1;
  required colorFilterSettings colorFilter = 2;
  required findPolySettings    findPoly = 3;
  required int32 	       videoOut =4;
}  
